doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+# Popup
+
+- Spawns a buffer inside a popup
+- You can use `vim.api` to manipulate it like any other buffer
+
+## Properties & Usage
+
+```lua
+local Popup = require("nui.popup")
+
+local popup = Popup({
+    position = "20%",
+    size = {
+        width = 20,
+        height = 10,
+    },
+    enter = true,
+    focusable = true,
+    zindex = 10,
+    relative = "editor",
+    border = {
+        highlight = "MyHighlightGroup",
+        padding = {
+            top = 2,
+            bottom = 2,
+            left = 3,
+            right = 3,
+        },
+        style = "double",
+        text = {
+            top = " I am top title ",
+            top_align = "center",
+            bottom = "I am bottom title",
+            bottom_align = "left",
+        },
+    },
+    buf_options = {
+        modifiable = true,
+        readonly = false,
+    },
+    win_options = {
+        winblend = 10,
+        winhighlight = "Normal:Normal",
+    },
+)
+```
+
+## Property Descriptions
+
+### `border`
+
+- **Type:** `table`
+- **Description:** Contains all border related options
+
+#### `padding`
+
+- **Type:** `table`
+- **Description:**
+    - Controls the popup padding
+    - Behaves like [CSS padding](https://developer.mozilla.org/en-US/docs/Web/CSS/padding)
+
+##### Usage
+
+```lua
+border = {
+    padding = { top = 10, bottom = 20, left = 15, right = 20 }
+}
+```
+
+```lua
+border = {
+    padding = { 10, 30, 40, 10 }
+}
```

Comment:
- Can we call these "Examples" instead of "Usage"?
- Nesting markdown headers so deep seems unnecessary? Maybe we can just do `**Examples**` instead?

Reply:
Sure thing.

The reason I was using markdown headers a lot is I have markdownlint enabled and as far as I know it "enforces" some standard markdown guidelines and those are don't bold something that should represent a title.

Thats the only reason pretty much and it is kinda hard to read so I'm down to switch back to bolded.